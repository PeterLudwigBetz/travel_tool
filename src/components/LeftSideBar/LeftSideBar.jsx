import React, {Component} from 'react';
import LeftSideNavItems from './LeftSideNavItems/LeftSideNavItems';
import activeBookmarkIcon from '../../images/icons/bookmark_active.svg';
import inactiveBookmarkIcon from '../../images/icons/bookmark_inactive.svg';
import activeSettingsIcon from '../../images/icons/settings_active.svg';
import inactiveSettingsIcon from '../../images/icons/settings_inactive.svg';
import './LeftSideBar.scss';

class LeftSideBar extends Component {

  state = {
    activeNavItem: {}
  };

  setActiveNavItem = (clickedNavItem) => {
    this.setState({activeNavItem: clickedNavItem});
  }

  getNavigationIcons = () => {
    return {
      requestsIcon: {
        active: activeBookmarkIcon,
        inactive: inactiveBookmarkIcon
      },
      settingsIcon: {
        active: activeSettingsIcon,
        inactive: inactiveSettingsIcon
      },
    };
  }

  render() {
    const {activeNavItem} = this.state;

    return (
      <div className="left-sidebar">
        <div className="left-sidebar__fixed_wrapper">
          <div style={{overflowY: 'scroll', height: '100%'}}>
            <LeftSideNavItems
              activeNavItem={activeNavItem}
              navIconsSource={this.getNavigationIcons()}
              setActiveNavItem={this.setActiveNavItem} />
          </div>
        </div>
      </div>
    );
  }
}

export default LeftSideBar;
