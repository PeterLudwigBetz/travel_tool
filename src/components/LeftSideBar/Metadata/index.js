import activeBookmarkIcon from '../../../images/icons/bookmark_active.svg';
import inactiveBookmarkIcon from '../../../images/icons/bookmark_inactive.svg';
import activeSettingsIcon from '../../../images/icons/settings_active.svg';
import inactiveSettingsIcon from '../../../images/icons/settings_inactive.svg';
import inactiveLogoutIcon from '../../../images/icons/logout_inactive.svg';
import activeAccomodationIcon from '../../../images/icons/accommodation-blue.svg';
import inactiveAccomodationIcon from '../../../images/icons/accomodation-grey.svg';

const NavItemsMetadata = [
  // Requests
  {
    text: 'Requests',
    link_to: '/requests',
    activateOnLogin: true,
    isDropdown: true,
    dropdownItems: [
      {
        link_to: '/requests',
        text: 'My Requests',
        exact: true
      },
      {
        link_to: '/requests/my-approvals',
        text: 'My Approvals'
      }
    ],
    icons: {
      active: activeBookmarkIcon,
      inactive: inactiveBookmarkIcon
    }
  },
  // Accommodation
  {
    text: 'Residence',
    link_to: '/residence',
    activateOnLogin: false,
    isDropdown: true,
    dropdownItems: [
      {
        link_to: '/residence',
        text: 'Manage',
        onlyVisibleTo: ['Travel Administrator', 'Super Administrator'],
        exact: true
      },
      {
        link_to: '/residence/checkin',
        text: 'Check-in'
      }
    ],
    icons: {
      active: activeAccomodationIcon,
      inactive: inactiveAccomodationIcon
    }
  },
  // Settings
  {
    text: 'Settings',
    link_to: '/settings',
    activateOnLogin: false,
    isDropdown: true,
    dropdownItems: [
      {
        link_to: '/settings/roles',
        text: 'User Roles',
        onlyVisibleTo: ['Super Administrator', ]
      },
      {
        link_to: '/settings/profile',
        text: 'User Profile',
      }

    ],
    icons: {
      active: activeSettingsIcon,
      inactive: inactiveSettingsIcon
    }
  },
  // Logout
  {
    text: 'Logout',
    link_to: '/logout',
    activateOnLogin: false,
    isDropdown: false,
    onClick: 'signout', // define a handler with this name in 'LeftSideNavItems'
    variantClassName: 'logout-button', // variant styling class
    icons: {
      inactive: inactiveLogoutIcon
    }
  },
];

export default NavItemsMetadata;
