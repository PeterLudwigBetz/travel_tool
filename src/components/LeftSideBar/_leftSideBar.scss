.sidebar {
  position: relative;
  height: 100vh;
}

.side-drawer .left-sidebar {

  &__fixed_wrapper {
    top: 65px;
    width: 100%;
  }
}

.left-sidebar {
  position: relative;
  top: 0px;
  box-sizing: border-box;
  margin: 0px;
  width: 100%;
  height: 100vh;
  font-family: 'DIN Pro';
  border-right: 1px solid #E4E4E4;

  &__fixed_wrapper {
    position: fixed;
    top: 0px;
    width: 16.666667%;
    height: 100vh;
  }

  &__scrollable_wrapper {
    overflow-y: scroll;
    height: 100%;
    position: relative;
    top: 108px;
  }

  .logout-button {
    @media(min-width: 838px) {
      display: none;
    }
  }
}
