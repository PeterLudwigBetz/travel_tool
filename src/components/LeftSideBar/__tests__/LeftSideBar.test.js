import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import LeftSideBar from '../index';

describe('<LeftSideBar />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<LeftSideBar />);
    expect(wrapper).toMatchSnapshot();
  });

  it('changes active navigation link', () => {
    const wrapper = mount(
      <MemoryRouter>
        <LeftSideBar />
      </MemoryRouter>
    );

    const defaultActiveNavItem = {};
    expect(wrapper
      .find(LeftSideBar)
      .instance().state.activeNavItem
    ).toEqual(defaultActiveNavItem);

    wrapper.find('.nav-link').at(0).simulate('click');
    const newState = wrapper.find(LeftSideBar).instance().state;
    expect(newState.activeNavItem).not.toEqual(defaultActiveNavItem);
  });
});
