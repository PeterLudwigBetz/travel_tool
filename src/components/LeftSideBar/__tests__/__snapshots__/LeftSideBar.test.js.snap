// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeftSideBar /> renders correctly 1`] = `
<div
  className="left-sidebar"
>
  <div
    className="left-sidebar__fixed_wrapper"
  >
    <div
      className="left-sidebar__scrollable_wrapper"
    >
      <withRouter(LeftSideNavItems)
        activeNavItem={Object {}}
        metadata={
          Array [
            Object {
              "activateOnLogin": true,
              "dropdownItems": Array [
                Object {
                  "exact": true,
                  "link_to": "/requests",
                  "text": "My Requests",
                },
                Object {
                  "link_to": "/requests/my-approvals",
                  "text": "My Approvals",
                },
              ],
              "icons": Object {
                "active": "bookmark_active.svg",
                "inactive": "bookmark_inactive.svg",
              },
              "isDropdown": true,
              "link_to": "/requests",
              "text": "Requests",
            },
            Object {
              "activateOnLogin": false,
              "dropdownItems": Array [
                Object {
                  "exact": true,
                  "link_to": "/residence",
                  "onlyVisibleTo": Array [
                    "Travel Administrator",
                    "Super Administrator",
                  ],
                  "text": "Manage",
                },
                Object {
                  "link_to": "/residence/checkin",
                  "text": "Check-in",
                },
              ],
              "icons": Object {
                "active": "accommodation-blue.svg",
                "inactive": "accomodation-grey.svg",
              },
              "isDropdown": true,
              "link_to": "/residence",
              "text": "Residence",
            },
            Object {
              "activateOnLogin": false,
              "dropdownItems": Array [
                Object {
                  "link_to": "/settings/roles",
                  "onlyVisibleTo": Array [
                    "Super Administrator",
                  ],
                  "text": "User Roles",
                },
                Object {
                  "link_to": "/settings/profile",
                  "text": "User Profile",
                },
              ],
              "icons": Object {
                "active": "settings_active.svg",
                "inactive": "settings_inactive.svg",
              },
              "isDropdown": true,
              "link_to": "/settings",
              "text": "Settings",
            },
            Object {
              "activateOnLogin": false,
              "icons": Object {
                "inactive": "logout_inactive.svg",
              },
              "isDropdown": false,
              "link_to": "/logout",
              "onClick": "signout",
              "text": "Logout",
              "variantClassName": "logout-button",
            },
          ]
        }
        setActiveNavItem={[Function]}
        userRole=""
      />
    </div>
  </div>
</div>
`;
